### DISCUSSION TITLE
How to perform cumulative sum based on rank?

### DISCUSSION THREAD

**Discussion by HuifangYeo** [atoti/atoti#785] [Q&A] [answered]
Hi,

I'm trying to perform cumulative summation along the rank that I have computed. How can I do that?
```
m["TotalProfit"] = tt.parent_value(m["Profit"], degrees={h["Products"]: 1})
m["%ProfitContribution"] = m["Profit"] / m["TotalProfit"]
m["ProfitContributionRank"] = tt.rank(
    m["%ProfitContribution"], h["Products"], ascending=False
)

m["%ProfitContribution"].formatter = "DOUBLE[0.00%]"

# m["%CumulativeProfitContribution"] = tt.agg.sum(
#     m["%ProfitContribution"], scope=tt.CumulativeScope(level=l[("Products", "Products", "ProductName")])
# )
```

![image](https://github.com/atoti/atoti/assets/35210495/9b23e5d0-eb2e-45d7-850e-5feef92efc18)

Sample notebook: [cumulative.zip](https://github.com/atoti/atoti/files/11437755/cumulative.zip)

**Comment by tibdex** [comment, marked as answer]
I don't see a way to do it apart from the one you're already using: declare **%CumulativeProfitContribution** like you did, sort by **Profit** (or **%ProfitContribution**), and don't rely on `atoti.rank()`.
